source/Octopus.Tentacle/Configuration/Instances/ApplicationInstanceSelector.cs

```diff
@@ -64,30 +69,48 @@ ApplicationInstanceConfiguration LoadCurrentInstance()
         ApplicationInstanceConfiguration LoadInstance()
         {
             var appInstance = LocateApplicationPrimaryConfiguration();
-            EnsureConfigurationExists(appInstance.instanceName, appInstance.configurationpath);
 
-            log.Verbose($"Loading configuration from {appInstance.configurationpath}");
-            var writableConfig = new XmlFileKeyValueStore(fileSystem, appInstance.configurationpath);
-            
-            var aggregatedKeyValueStore = ContributeAdditionalConfiguration(writableConfig);
+            var (aggregatedKeyValueStore, writableConfig) = LoadConfigurationStore(appInstance);
+
             return new ApplicationInstanceConfiguration(appInstance.instanceName, appInstance.configurationpath, aggregatedKeyValueStore, writableConfig);
         }
 
-        void EnsureConfigurationExists(string? instanceName, string configurationPath)
+        (IKeyValueStore, IWritableKeyValueStore) LoadConfigurationStore((string? instanceName, string? configurationpath) appInstance)
         {
-            if (!fileSystem.FileExists(configurationPath))
+            if (appInstance is { instanceName: not null, configurationpath: null } &&
+                KubernetesConfig.Namespace is {} @namespace)
             {
-                var message = !string.IsNullOrEmpty(instanceName)
-                    ? $"The configuration file for instance {instanceName} could not be located at the specified location {configurationPath}. " +
-                    "The file might have been manually removed without properly removing the instance and as such it is still listed as present." +
-                    "The instance must be created again before you can interact with it."
-                    : $"The configuration file at {configurationPath} could not be located at the specified location.";
-                    
-                throw new ControlledFailureException(message);
+                log.Verbose($"Loading configuration from ConfigMap for namespace {@namespace}");
+                var configMapWritableStore = configMapStoreFactory.Value;
+                return (ContributeAdditionalConfiguration(configMapWritableStore), configMapWritableStore);
             }
```

Comment:
This check seems weird (checking the namespace). Should we be using the `PlatformDetection` here?

Reply:
Ah yep, good call!

Reply:
Fixed 👍